fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -459,20 +465,27 @@ public static SearchParameter getSearchParameter(Class<?> resourceType, Canonica
      */
     public static SearchParameter getSearchParameter(String resourceType, Canonical uri) throws Exception {
         String tenantId = FHIRRequestContext.get().getTenantId();
-
+        SearchParameter result = null;
+        
         // First try to find the search parameter within the specified tenant's map.
-        SearchParameter result =
-                getSearchParameterInternal(getTenantOrDefaultSPMap(tenantId), resourceType, uri.getValue(), false);
+        Map<String, ParametersMap> spMaps = getTenantOrDefaultSPMap(tenantId);
+        if (spMaps != null && !spMaps.isEmpty()) {
+            ParametersMap parametersMap = spMaps.get(resourceType);
```

Comment:
473 and 481 have nearly identical blocks.

Reply:
I wasn't sure if factoring these couple lines into a separate method would be worth it, but I did it and I agree it cleans up the method a bit.